src/Strategy/Maven/Common.hs

```diff
@@ -22,6 +25,43 @@ data MavenDependency = MavenDependency
 mavenDependencyToDependency :: MavenDependency -> Dependency
 mavenDependencyToDependency MavenDependency{..} = dependency
 
+filterMavenSubmodules :: Set Text -> Set Text -> Graphing MavenDependency -> (Graphing MavenDependency)
+filterMavenSubmodules includedSubmoduleSet completeSubmoduleSet graph = do
+  let excludedSubmoduleSet = completeSubmoduleSet `Set.difference` includedSubmoduleSet
+      excludedSubmodules = Set.fromList $ Prelude.filter (\dep -> depNameFromMavenDependency dep `Set.member` excludedSubmoduleSet) $ vertexList graph
+      -- Find all the dependencies of the submodules that will be included in the scan
+      includedSubmoduleDependencies = combinedMavenDependencySet includedSubmoduleSet
```

Comment:
[Prefer] I think it's better if we write generic colouring utility function in Graphing module for this (some of these might already exists). 
There is also this: https://github.com/fossas/fossa-cli/blob/master/docs/contributing/graph-hydration.md which you might be able to use (refer to our npm strategy / package.json for usage example) 

Maybe some thing like this? (very rough sketch structure that came to mind/ may have to change signatures to compile/use)
```haskell
-- Adds color for all reachable children of 'a'.
color :: Graphing (a, Set a) -> a -> Graphing  (a, Set a)
color =

-- All children nodes at any depth originating from 'a', including 'a'
reachableFrom :: Graphing a -> a -> [a]
reachableFrom = -- this or similar may already in exist Graphing
```

then we should be able to simplify biz logic for readability with maybe something like:

```haskell
-- 0. Identify all submodule dependency, since
-- we only have name of submodule, which should be same as name of dependency 
subMods = ....

-- 1. Prepare for origin tagging as color
g = gmap (\a -> (a, mempty)) graph

-- 2. For each sub module, we color all reachable nodes from subMods
for subMod in subMods: -- may have to use haskell loops/fold/etc.
    g = color g subMod

-- 3. We include all nodes which were colored for any member of include submodules
-- If I'm reading code right, we don't really care for any deps which were in excluded set, so
-- shared will be automatically included, I'm presuming this is intended behaviour anyways
g = filter (hasIncludeMod) g

-- 4. Discard coloring
g = gmap (\(a, colors) -> a) g 

withoutSubModAsDep g

where
   -- True if includeMod is colored, false Otherwise.
   hasIncludeMod :: (Dependency, Set Dependency) -> Bool
   hasIncludeMod (dep, origins) = .... -- null $ (Set.map nameOf origin) `intersection`  (Set.map nameOf includedSubmoduleSet)
   
   withoutSubModAsDep :: Graphing Dependency -> Graphing Dependency
   withoutSubModAsDep = filter isSubMod
   
   isSubMod :: Dependency -> Bool
   isSubMod dep = ... 
 
 
```